storage/innobase/xtrabackup/src/ds_fifo.cc

```diff
@@ -35,6 +35,28 @@ typedef struct {
   std::unordered_map<std::string, File> FIFO_list;
   /* Mutex protecting FIFO list */
   std::mutex fifo_mutex;
+
+  /* Add a new pair of fullpath and fd to FIFO_list */
+  void populate_list(std::string fullpath, File fd) {
```

Comment:
we now have methods, so we can make this as class, make members private and add a destructor that closes fd? Where do we return FIO_list objects back to map?

Reply:
They are returned at fifo_close:
```
static int fifo_close(ds_file_t *file) {
  . . .
  fifo_context->populate_list(((ds_fifo_file_t *)file)->path, fd);
```

Reply:
> and add a destructor that closes fd?
This is what fifo_close does. https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2514-8.0/storage/innobase/xtrabackup/src/ds_fifo.cc#L183 
The thing is that all datasink have a specific contract they have to implement which is defined at datasink.cc .

Reply:
I can convert this to a class but I don't see much benefit from deviant from original datasink standard

Reply:
ok, np  it is not super important anyway.